specs/typography/typography.md

```diff
@@ -0,0 +1,143 @@
+
+![](./images/type-ramp.png)
+
+Note the "Ramp ID", which is used to select the desired size. This ramp is centered around the base type size to associate the relative sizes between content. For instance, your main body text uses the `base` size. You may want a caption to be smaller than that, so you use `-1`. You want your [headings](./headings.md) to be larger, so you use one or more of the `+` sizes.
+
+This is more declarative than using an index position to reference the sizes, like `7` if this ramp is numbered from the top or `3` if numbered from the bottom. It also allows the sizes that you use to be related to each other rather than at other arbitrary points on the ramp. For example, three levels of headings above body text are `16px`, `20px`, and `28px` instead of jumping all the way to `60px` at the top and skipping the middle. You can add additional sizes at the top and bottom if necessary.
+
+*Note that these are just examples and nothing about any particular usage of a type size is prescribed by the underlying ramp. See further typography specs for other systematic ways to apply the sizes.*
+
+Type style variables can be added at any point in the DOM. Generally they would be created globally for consistent application throughout your app or site, or they could be overridden or added for a single section of a page.
+
+### API
+
+Pseudo interfaces: (*Could be functions or decorators or otherwise*)
+
+- `typeRamp()`: A default type ramp as illustrated above
+- `typeRamp({'12px', '14px', '24px', '36px'}, 1)`: A type ramp with fixed type size values and default line height, index 1 is base
+- `typeRamp({{'12px', '16px'}, {'14px', '20px'}, {'24px', '28px'}, {'36px', '44px'}}, 1)`: A type ramp with fixed type size values and fixed individual line heights
+- `typeRamp(16, 1.25, 3, 5)`: A type ramp based at `16px` on a major third [modular scale](https://type-scale.com/) of `1.25` with `3` sizes below base and `5` sizes above.
+
+### Anatomy and Appearance
+
+The type ramp is presented as css variables that extend plus and minus from the base.
+
+- `fast-type-ramp-base-size` = `"14px"`
+- `fast-type-ramp-base-height` = `"20px"`
+- `fast-type-ramp-plus-1-size` = `"16px"`
+- `fast-type-ramp-plus-1-height` = `"24px"`
+- `fast-type-ramp-plus-2-size` = `"20px"`
+- `fast-type-ramp-plus-2-height` = `"28px"`
+- `fast-type-ramp-plus-3-size` = `"28px"`
+- `fast-type-ramp-plus-3-height` = `"36px"`
+- `fast-type-ramp-plus-4-size` = `"34px"`
+- `fast-type-ramp-plus-4-height` = `"44px"`
+- `fast-type-ramp-plus-5-size` = `"46px"`
+- `fast-type-ramp-plus-5-height` = `"56px"`
+- `fast-type-ramp-plus-6-size` = `"60px"`
+- `fast-type-ramp-plus-6-height` = `"72px"`
+- `fast-type-ramp-minus-1-size` = `"12px"`
+- `fast-type-ramp-minus-1-height` = `"16px"`
+- `fast-type-ramp-minus-2-size` = `"10px"`
+- `fast-type-ramp-minus-2-height` = `"16px"`
```

Comment:
nit and more of an organizational change; based on the idea that these values scale out from the middle, it might help people visualize the "ramp" part of it by mimicking that here.

```
minus
base
plus
```

Reply:
As an aside - does this start from the "middle"? It looks like there are more plus values than minus values.

Reply:
No, the base is not necessarily the middle of the ramp, in fact probably rarely. I described it as extending up and down from the base, but does it read as "middle"?

Reply:
Ah, I see "centered". I'll update that.

Reply:
I think the primary thing I saw was that there seems to be "plus six" but I only saw "minus two". It could just be wording, but it makes it seem like the ramp looks like this:

```
-2
-1
Base
+1
+2
+3
+4
+5
+5
```

Reply:
Right, that's the intent. The base size is not in the middle, and generally there will probably be more larger sizes than smaller.